### DISCUSSION TITLE
Setting up Zustand for testing with RTL – comment about uncurried create

### DISCUSSION THREAD

**Discussion by skagedal** [pmndrs/zustand#2471] [Q&A]
Hello! On the [Testing](https://docs.pmnd.rs/zustand/guides/testing) page in the docs, there is this comment:

> **Warning:** In the following examples we are only supporting the curried version of create (`create()(...)`), since it is supported on both JavaScript and TypeScript. If you are using the uncurried version of `create(...)` you will need to update your hooks to use the curried version, or update the mocks in order to support the uncurried version.

However, when I look at the code under Jest for example, it looks to me as it is handling both the curried and uncurried versions:

```ts
  // to support curried version of create
  return typeof stateCreator === 'function'
    ? createUncurried(stateCreator)
    : createUncurried
```

In our project, we are using uncurried create calls, and it does seem to work. Is the comment wrong (maybe not up-to-date with the code examples), or am I misunderstanding something about what's not supposed to work?

**Comment by dbritto-dev** [comment]
@skagedal the code should work for both but you can use the one from the docs. What error are you getting?

**Comment by skagedal** [comment]
@dbritto-dev I am not getting errors, I am asking about the documentation. Shouldn't this warning be removed, then?

**Reply by dbritto-dev** [reply]
@skagedal yes and no, the example was created focus on typescript and the demo just added some extra lines to support ts and js

**Reply by skagedal** [reply]
Sorry, I don't understand. Are the "example" and the "demo" two different things? Are the "following examples" mentioned in the warning not referring to the code snippets below?

**Reply by dbritto-dev** [reply]
@skagedal  my bad, you are right seems that someone updated the demo code and the example code without removing the warning added initially

**Reply by dbritto-dev** [reply]
@skagedal let me update it and remove it, and seems that those codesandbox links are not working anymore so I guess I need to recreate them on stackblitz

**Reply by skagedal** [reply]
Awesome, thanks!

**Reply by dbritto-dev** [reply]
@skagedal  here you go -> https://github.com/pmndrs/zustand/pull/2474

**Reply by skagedal** [reply]
Very nice!
